File: content/fixtures/code.go
package main

import "fmt"

type Checkpoint struct {
	Name   string
	Status string
	Detail string
}

func describe(checkpoint Checkpoint) string {
	return fmt.Sprintf("%s: %s (%s)", checkpoint.Name, checkpoint.Status, checkpoint.Detail)
}

func main() {
	checkpoints := []Checkpoint{
		{Name: "ingest", Status: "ok", Detail: "rows normalized"},
		{Name: "review", Status: "hold", Detail: "extraction fixture marker: shared code sample"},
		{Name: "publish", Status: "ok", Detail: "index refreshed"},
	}

	for _, checkpoint := range checkpoints {
		fmt.Println(describe(checkpoint))
	}
}

File: content/fixtures/code.php
<?php

declare(strict_types=1);

final class QueueEntry
{
    public function __construct(
        public readonly string $name,
        public readonly string $state,
        public readonly string $detail,
    ) {}
}

function describeEntry(QueueEntry $entry): string
{
    return sprintf('%s [%s]: %s', $entry->name, $entry->state, $entry->detail);
}

$entries = [
    new QueueEntry('parse', 'complete', 'front matter loaded'),
    new QueueEntry('extract', 'review', 'extraction fixture marker: shared code sample'),
    new QueueEntry('index', 'pending', 'search document queued'),
];

foreach ($entries as $entry) {
    echo describeEntry($entry) . PHP_EOL;
}
